redbot/cogs/mod/events.py

```diff
@@ -72,6 +71,60 @@ async def check_mention_spam(self, message):
                         print(e)
                         return False
                     return True
+
+        if mention_spam["kick"]:
+            if len(mentions) >= mention_spam["kick"]:
+                try:
+                    await guild.kick(author, reason=_("Mention Spam (Autokick)"))
+                except discord.HTTPException:
+                    log.info(
+                        "Failed to kick member for mention spam in server {}".format(guild.id)
+                    )
+                else:
+                    try:
+                        await modlog.create_case(
+                            self.bot,
+                            guild,
+                            message.created_at,
+                            "kick",
+                            author,
+                            guild.me,
+                            _("Mention spam (Autokick)"),
+                            until=None,
+                            channel=None,
+                        )
+                    except RuntimeError as e:
+                        print(e)
+                        return False
+                    return True
+
+        if mention_spam["warn"]:
+            if len(mentions) >= mention_spam["warn"]:
+                try:
+                    await author.send(
+                        _("Please do not mention {} or more people!".format(mention_spam["warn"]))
```

Comment:
I'm not sure stating the warn number is the *best* idea, it's not always a good idea to let users know exactly what trips the bots in a server (beyond knowing something is against the rules in general).